mongodb/src/main/scala/akka/stream/alpakka/mongodb/javadsl/MongoSink.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mongodb.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.{Done, NotUsed}
+import akka.stream.alpakka.mongodb.DocumentUpdate
+import akka.stream.alpakka.mongodb.scaladsl.MongoFlow.{
+  DefaultDeleteOptions,
+  DefaultInsertManyOptions,
+  DefaultInsertOneOptions,
+  DefaultUpdateOptions
+}
+import akka.stream.javadsl.{Keep, Sink}
+import com.mongodb.client.model.{DeleteOptions, InsertManyOptions, InsertOneOptions, UpdateOptions}
+import com.mongodb.reactivestreams.client.MongoCollection
+import org.bson.conversions.Bson
+
+object MongoSink {
+
+  /**
+   * A [[akka.stream.javadsl.Sink Sink]] that will insert documents into a collection.
+   *
+   * @param collection mongo db collection to insert to.
+   */
+  def insertOne[T](collection: MongoCollection[T]): Sink[T, CompletionStage[Done]] =
+    insertOne(collection, DefaultInsertOneOptions)
+
+  /**
+   * A [[akka.stream.javadsl.Sink Sink]] that will insert documents into a collection.
+   *
+   * @param collection mongo db collection to insert to.
+   * @param options options to apply to the operation
+   */
+  def insertOne[T](collection: MongoCollection[T], options: InsertOneOptions): Sink[T, CompletionStage[Done]] =
```

Comment:
We could maybe drop the alternatives with options from the sink API. If users need those, they may use the Flow API. WDYT?

Reply:
I looked at it again, and the Sink variants are used quite extensively in tests. I say the shrinkage of the API surface is not worth the introduced discrepancy.